### PR TITLE
perf: optimize get tso frequency

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
Querying CurrentVersion too frequently when resolving lock. #790 

### What is changed and how it works?
CurrentVersion is now cached and is updated regularly by a separate goroutine.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes
 - Has exported function/method change

### Release note

- No release note

### CODE REVIEW

File: cdc/kv/client.go [hunk 1]
```diff
@@ -251,6 +252,8 @@ type CDCClient struct {
 
 	regionCache *tikv.RegionCache
 	kvStorage   tikv.Storage
+
+	currentVersionCache tidbkv.Version
```
Comment (leoppro) [inline cdc/kv/client.go:256]:
```suggestion
	currentVersionCache model.Ts
```

File: cdc/kv/client.go [hunk 2]
```diff
@@ -270,7 +273,26 @@ func NewCDCClient(ctx context.Context, pd pd.Client, kvStorage tikv.Storage, cre
...
 	}
+
+	go (func() {
+		ticker := time.NewTicker(tsoQueryInterval)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
```
Comment (leoppro) [inline cdc/kv/client.go:286]: if there is no region to resolve lock, we still query the tso from pd every two seconds?

File: cdc/kv/client.go [hunk 3]
```diff
@@ -286,6 +308,14 @@ func (c *CDCClient) Close() error {
 	return nil
 }
 
+// GetCurrentVersion gets a cached version that is regularly updated by a background goroutine
+func (c *CDCClient) GetCurrentVersion() tidbkv.Version {
+	for atomic.LoadUint64(&c.currentVersionCache.Ver) == 0 {
+		time.Sleep(1 * time.Second)
```
Comment (leoppro) [inline cdc/kv/client.go:314]: It's not graceful

File: cdc/kv/client.go [hunk 4]
```diff
@@ -286,6 +308,14 @@ func (c *CDCClient) Close() error {
 	return nil
 }
 
+// GetCurrentVersion gets a cached version that is regularly updated by a background goroutine
+func (c *CDCClient) GetCurrentVersion() tidbkv.Version {
```
Comment (overvenus) [inline cdc/kv/client.go:312]: How many CDCClient would CDC create? How about caching the TS in the processor?

File: cdc/puller/puller.go [hunk 5]
```diff
@@ -59,6 +61,101 @@ type pullerImpl struct {
...
+				// the cached value has been updated once
+				ver = atomic.LoadUint64(&s.currentVersionCache)
+				break loop
+			}
+		}
+	}
+	return tidbkv.Version{Ver: ver}, nil
+}
+
+func (s *kvStorageWithVersionCache) Close() error {
```
Comment (amyangfei) [inline cdc/puller/puller.go:136]: Where is the `Close` called

File: cdc/puller/puller.go [hunk 6]
```diff
@@ -15,6 +15,7 @@ package puller
 
 import (
 	"context"
+	"github.com/pingcap/ticdc/pkg/notify"
```
Comment (amyangfei) [inline cdc/puller/puller.go:18]: Please move this import to the group of third-party lib

File: cdc/puller/puller.go [hunk 7]
```diff
@@ -69,6 +166,7 @@ func NewPuller(
 	spans []regionspan.Span,
 	limitter *BlurResourceLimitter,
 ) Puller {
+	kvStorage = newKvStorageWithVersionCache(kvStorage)
```
Comment (leoppro) [inline cdc/puller/puller.go:169]: `KvStorageWithVersionCache` should be globally unique, not one per puller.

**Pull request merged**